### PR TITLE
[shardformer]support gradients accumulation fusion

### PR DESCRIPTION
**Pull request by flybird11111**
## 📌 Checklist before creating the PR

- [x] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

#5198 

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

support gradients accumulation fusion, the current gradient is accumulated onto the previously computed gradient, all done in a single operation, which helps avoid multiple accesses to global memory and enhances the operator's bandwidth.

## 💥 Checklist before requesting a review

- [x] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [ ] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: colossalai/shardformer/layer/linear.py [hunk 1]
```diff
@@ -408,7 +408,7 @@ def forward(self, input_: Tensor) -> Tensor:
                     handle.wait()
                 output = torch.cat(output_parallel_list, dim=-1)
         else:
-            output_parallel = F.linear(input_, self.weight)
+            output_parallel = linear_with_async_comm(input_, self.weight, None, None, False)
```
[the remark this answers is not in the document]
Comment (flybird11111) [reply colossalai/shardformer/layer/linear.py:411]: Linear1D_Row can also use fused_weight_gradient_mlp_cuda, and `linear_with_async_comm` has implemented this functionality. By setting async_grad_allreduce to false, I replaced F.linear in Linear1D_Row, eliminating the need to introduce an additional operator for Linear1D_Row.

**Review by ver217 — APPROVED** [review]

**Comment by flybird11111** [comment]
image
image
All tests have passed.
